codex/blockexchange/engine/discovery.nim

```diff
@@ -102,7 +102,7 @@ proc advertiseQueueLoop(b: DiscoveryEngine) {.async.} =
       trace "Begin iterating blocks..."
       for c in cids:
         if cid =? await c:
-          b.advertiseBlock(cid)
+          asyncSpawn b.advertiseBlock(cid)
```

Comment:
Why `asyncSpawn`?

Reply:
If you check how the code was before, this call to `advertiseBlock` actually had nothing before it. I'm not even sure how this was compiling. `asyncspawn` is the closest thing in terms of semantics.

Reply:
Hmm, that is weird... 

`advertiseBlock` pushes the blocks into an async queue with a limited number of entries, this is done for backpressure purposes. An `asyncSpawn` will effectively invalidate the queues backpressure. The correct thing to do is to await it and the `asyncSleep` is most likely unneeded, but probably harmless for now.

Reply:
Well in this case let's fix it then! :-)